cloudinit/distros/bsd.py

```diff
@@ -149,3 +150,23 @@ def get_proc_ppid(pid):
         """
         ppid, _ = subp.subp(["ps", "-oppid=", "-p", str(pid)])
         return int(ppid.strip())
+
+    @staticmethod
+    def get_mapped_device(blockdev: str) -> Optional[str]:
+        return None
+
+    @staticmethod
+    def device_part_info(devpath: str) -> tuple:
+        # FreeBSD doesn't know of sysfs so just get everything we need from
+        # the device, like /dev/vtbd0p2.
+        part = util.find_freebsd_part(devpath)
```

Comment:
this function is FreeBSD (and Dragonfly) specific.
This is the BSD base-class, and should work on all OSes;
I'm not sure we have such a thing, but might be able to cobble it together.

Reply:
+1 If you don't mind helping that would be appreciated. I believe that this code works as it did before, since this is what would get called if `util.is_BSD()` returned true.

Reply:
in trying to patch this up, I realied that right now, we don't actually support growpart/growfs on NetBSD or OpenBSD, so, yeah, right now, that code will work just as before. OpenBSD at least doesn't support online resizing, so it's no big deal there.

I think we can go forth as is, and then when NetBSD people finally complain we can start fixing it.
meanwhile, I'll build an image to test growpart on FreeBSD